cluster/cluster_interface.ml

```diff
@@ -255,4 +255,10 @@ module LocalAPI (R : RPC) = struct
     declare "get-config"
       ["Returns local cluster config"]
       (debug_info_p @-> returning cluster_config_p err)
+
+  let write_pem =
+    let pems_p = Param.mk ~name:"pems" pems in
+    declare "write-pem"
```

Comment:
might we worth calling it 'write-pems' as it can write a list, not just one. How would this function interact with a hypothetical certificate rotation for the cluster?

Reply:
I will rename it to write-pems :+1: 

To make 'cluster cert rotation' work you could probably do something like this
 - xapi calls 'get-config' to retrieve the current pem and cn from the cluster
 - xapi calls 'write-pems' with `{cn; blobs=[current_pem; new_pem]}`
 - xapi calls `write-pems' with `{cn; blobs=[new_pem]}`

(we'd also need to add a flag in write-pems which means  'write the new pems to disk and start using them' - currently all 'write-pems' does is adds new pems to the cluster config, such that the pems will be used once the daemon next restarts)

An alternative approach is to add a 'rotate_pems' API, and leave the entire implementation to clusterd (which is probably cleaner)

It's worth noting that the current model doesn't allow for a change of `cn`. I don't believe this is necessary however